### PR TITLE
Allow null entity ID in contract result response

### LINKED ISSUE #8430: Issue with Transactions Resulting in `INVALID_ACCOUNT_ID` Errors
### Description

This transaction, https://hashscan.io/testnet/transaction/1717032617.100851448, on hashscan it says that this transaction failed because of INVALID_ACCOUNT_ID. Then within the network tab of the browser of this same transaction, I found a mirror-node request sent to contracts/results endpoint like this:
```bash
https://testnet.mirrornode.hedera.com/api/v1/contracts/results?timestamp=1717032617.100851448&internal=true
```
That request returns
```bash
{
  "_status": {
    "messages": [
      {
        "message": "Null entity ID"
      }
    ]
  }
}
```
with a Status Code of `400 Bad Request`.

Now, in this next transaction, https://hashscan.io/testnet/transaction/1717032619.820208990, which is also a failure transaction but says `WRONG_NONCE`. Then the mirror-node request sent to contracts/results endpoint like this:
```bash
https://testnet.mirrornode.hedera.com/api/v1/contracts/results?timestamp=1717032619.820208990&internal=true
```
That request returns
```bash
{
  "results": [],
  "links": {
    "next": null
  }
}
```
with a Status code of `200 OK`.

why does `INVALID_ACCOUNT_ID` / `Null entity ID` error throw `400 Bad Request`, but `WRONG NONCE` works fine?

This issue is currently the main block of this ticket https://github.com/hashgraph/hedera-json-rpc-relay/issues/2557 on the Relay repo. The request to the mirror node throws a Null entity ID error when attempting to retrieve the contract result of the transaction by sending a mirror node request to contracts/result, causing the entire request to fail.

### Additional context

_No response_

### Hedera network

mainnet, testnet, previewnet, other

### Version

latest

### Operating system

None

### PR DESCRIPTION
**Pull request by jnels124**
**Description**:

Allow sender in contract result response to be null

**Related issue(s)**:

Fixes #8430

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/viewmodel/contractResultViewModel.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ class ContractResultViewModel {
     this.contract_id = contractId.toString();
     this.created_contract_ids = _.toArray(contractResult.createdContractIds).map((id) => EntityId.parse(id).toString());
     this.error_message = _.isEmpty(contractResult.errorMessage) ? null : contractResult.errorMessage;
-    this.from = EntityId.parse(contractResult.senderId).toEvmAddress();
+    this.from = EntityId.parse(contractResult.senderId, {isNullable: true}).toEvmAddress();
```
Comment (jnels124) [inline hedera-mirror-rest/viewmodel/contractResultViewModel.js:41]: could be more restrictive and ensure we only allow null for appropriate transaction results

**Comment by jnels124** [comment]
adding additional tests

**Review by steven-sheehy — DISMISSED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**